src/Picasso_UniformMesh.hpp

```diff
@@ -214,13 +255,15 @@ struct is_uniform_mesh
 
 //---------------------------------------------------------------------------//
 // Creation function.
-template <class MemorySpace>
+template <class MemorySpace, class ExecSpace>
 auto createUniformMesh( MemorySpace, const boost::property_tree::ptree& ptree,
                         const Kokkos::Array<double, 6>& global_bounding_box,
-                        const int minimum_halo_cell_width, MPI_Comm comm )
+                        const int minimum_halo_cell_width, MPI_Comm comm,
+                        ExecSpace )
 {
     return std::make_shared<UniformMesh<MemorySpace>>(
-        ptree, global_bounding_box, minimum_halo_cell_width, comm );
+        ptree, global_bounding_box, minimum_halo_cell_width, comm,
+        ExecSpace() );
```

Comment:
You could just pass an instance of the execution space on line 262 and then use that instance here.

Reply:
In fact this is preferred because the idea of using execution space instances is to encapsulate execution state (e.g. CUDA streams)